cmd/kubeless/update.go

```diff
@@ -72,9 +73,14 @@ var updateCmd = &cobra.Command{
 		if err != nil {
 			logrus.Fatal(err)
 		}
+		funcMem := resource.Quantity{}
+		if mem != "" {
+			funcMem = parseMemory(mem)
+		}
 
 		funcType := "HTTP"
-		err = utils.UpdateK8sCustomResource(runtime, handler, file, funcName, funcType, ns, description, mem, labels, envs)
+		f := constructFunction(runtime, handler, file, funcName, funcType, "", ns, "", description, funcMem, labels, envs)
```

Comment:
I think you misunderstood my suggestion, since you've just moved the ever-growing argument list to another function :(

I suggest you create the actual Function object right here, inline.  If you look at `constructFunction`, all it does is this, so you have saving any typing by parsing all your arguments into a long list of temporary variables, and then passing all those variables to a function that basically exists just to list them again in struct-literal form.  Just build the struct right here.

If `constructFunction` did something clever or difficult then it would be different - but it really is just copying arguments into a struct in the obvious way.

Alternatively, perhaps you could have a function that builds a `Function` from `cmd`, since this you want to support the exact same arguments for both create and update (I think)?  Mostly I want an approach that means we won't have a function that requires 20 moderately obscure arguments to be given in precisely the correct order in the near future.